lib/vmpooler/api/v1.rb

```diff
@@ -245,36 +307,85 @@ def sync_pool_templates
       pool_index = pool_index(pools)
       template_configs = backend.hgetall('vmpooler__config__template')
       template_configs&.each do |poolname, template|
-          if pool_index.include? poolname
-            unless pools[pool_index[poolname]]['template'] == template
-              pools[pool_index[poolname]]['template'] = template
-            end
-          end
+        next unless pool_index.include? poolname
+
+        pools[pool_index[poolname]]['template'] = template unless pools[pool_index[poolname]]['template'] == template
       end
     end
 
     def sync_pool_sizes
       pool_index = pool_index(pools)
       poolsize_configs = backend.hgetall('vmpooler__config__poolsize')
       poolsize_configs&.each do |poolname, size|
-          if pool_index.include? poolname
-            unless pools[pool_index[poolname]]['size'] == size.to_i
-              pools[pool_index[poolname]]['size'] == size.to_i
-            end
-          end
+        next unless pool_index.include? poolname
+
+        pools[pool_index[poolname]]['size'] == size.to_i unless pools[pool_index[poolname]]['size'] == size.to_i
```

Comment:
Similar question here to above, this seems very obscured to me.

Reply:
sync_pool_sizes is similar to syncing pool templates. If the pool size differs in the configuration file and in redis (from a previous API request changing the pool size) then on startup vmpooler would display the value in the configuration file. This method updates the configuration to reflect that the value loaded into redis take precedence over the value configured in the configuration file.

Reply:
This one is especially confusing because it's doing equality not assignment.

Reply:
Yes. I don't think that is what the intention was. I adjusted this so it is actually assigning data.